tests/testapp/factories.py

```diff
@@ -180,3 +182,20 @@ class Meta:
 class MiddlewareModelFactory(factory.django.DjangoModelFactory):
     class Meta:
         model = MiddlewareModel
+
+
+class RedirectFactory(factory.django.DjangoModelFactory):
+    """
+    TODO: Consider moving to wagtail factories?
+    """
+
+    old_path = factory.Faker("slug")
+    # Note: Site needs `hostname` and `port` to be a unique combination,
+    # so this seems like the best option:
+    site = factory.Iterator(Site.objects.all())
```

Comment:
While a nice idea, `wagtail-factories` contains a factory for Site's so we can just use that:

https://github.com/wagtail/wagtail-factories/blob/main/src/wagtail_factories/factories.py#L132

```suggestion
    site = factory.SubFactory(wagtail_factories.SiteFactory)```

Reply:
Addressed in 2822f362c4cfc4dfd11385da06ac38d630f1824a